sp-sim/src/lib.rs

```diff
@@ -50,6 +55,44 @@ pub trait SimulatedSp {
         &self,
         request: RotRequestV1,
     ) -> Result<RotResponseV1, RotSprocketError>;
+
+    /// Get the last completed update delivered to this simulated SP.
+    ///
+    /// Only returns data after a simulated reset.
+    async fn last_update_data(&self) -> Option<Box<[u8]>>;
+
+    /// Get the current update status, just as would be returned by an MGS
+    /// request to get the update status.
+    async fn current_update_status(&self) -> gateway_messages::UpdateStatus;
+
+    /// Get a watch channel on which this simulated SP will publish a
+    /// monotonically increasing count of how many responses it has successfully
+    /// sent.
+    ///
+    /// Returns `None` if called before the SP has set up its sockets to handle
+    /// requests.
+    fn responses_sent_count(&self) -> Option<watch::Receiver<usize>>;
+
+    /// Inject a UDP-level "throttler" on this simualted SP.
```

Comment:
This sounds like basically a semaphore.  Is that right?

Reply:
Yeah, I'll reword this. I think when I first wrote this I was using a tokio task to use this to throttle the SP to N responses/second, but it definitely is implementing a sempahore.

Reply:
Up to you.  I was just checking my understanding.